things/postgres/things.go

```diff
@@ -209,10 +205,15 @@ func (tr thingRepository) RetrieveAll(owner string, offset, limit uint64) things
 		},
 	}
 
-	return page
+	return page, nil
 }
 
-func (tr thingRepository) RetrieveByChannel(owner, channel string, offset, limit uint64) things.ThingsPage {
+func (tr thingRepository) RetrieveByChannel(owner, channel string, offset, limit uint64) (things.ThingsPage, error) {
+	// Verify if UUID format is valid to avoid internal Postgres error
+	if _, err := uuid.FromString(channel); err != nil {
```

Comment:
Why are we doing this UUID validation here and not at `transport` layer, or at some earlier stages?

Reply:
@anovakovic01 @nmarcetic @dusanb94 please also comment on this question ^

Reply:
Because if we ever change ID format (because of DB change or something else) we only have to change repository implementation.

Reply:
OK, thanks for clarification.